src/element/textElement.ts

```diff
@@ -714,34 +710,53 @@ export const getTextBindableContainerAtPosition = (
   return isTextBindableContainer(hitElement, false) ? hitElement : null;
 };
 
-export const isValidTextContainer = (element: ExcalidrawElement) => {
-  return (
-    element.type === "rectangle" ||
-    element.type === "ellipse" ||
-    element.type === "diamond" ||
-    isImageElement(element) ||
-    isArrowElement(element)
-  );
-};
+const VALID_CONTAINER_TYPES = [
+  "rectangle",
+  "ellipse",
+  "diamond",
+  "image",
+  "arrow",
+];
+
+export const isValidTextContainer = (element: ExcalidrawElement) =>
+  VALID_CONTAINER_TYPES.includes(element.type);
 
 export const computeContainerHeightForBoundText = (
-  container: NonDeletedExcalidrawElement,
   boundTextElementHeight: number,
+  containerType: typeof VALID_CONTAINER_TYPES[number],
 ) => {
-  if (container.type === "ellipse") {
+  if (containerType === "ellipse") {
     return Math.round(
       ((boundTextElementHeight + BOUND_TEXT_PADDING * 2) / Math.sqrt(2)) * 2,
     );
   }
-  if (isArrowElement(container)) {
+  if (containerType === "arrow") {
     return boundTextElementHeight + BOUND_TEXT_PADDING * 8 * 2;
   }
-  if (container.type === "diamond") {
+  if (containerType === "diamond") {
     return 2 * (boundTextElementHeight + BOUND_TEXT_PADDING * 2);
   }
   return boundTextElementHeight + BOUND_TEXT_PADDING * 2;
 };
 
+export const computeContainerWidthForBoundText = (
+  boundTextElementWidth: number,
+  containerType: typeof VALID_CONTAINER_TYPES[number],
+) => {
+  if (containerType === "ellipse") {
+    return Math.round(
+      ((boundTextElementWidth + BOUND_TEXT_PADDING * 2) / Math.sqrt(2)) * 2,
+    );
+  }
+  if (containerType === "arrow") {
+    return boundTextElementWidth + BOUND_TEXT_PADDING * 8 * 2;
+  }
+  if (containerType === "diamond") {
+    return 2 * (boundTextElementWidth + BOUND_TEXT_PADDING * 2);
+  }
+  return boundTextElementWidth + BOUND_TEXT_PADDING * 2;
+};
+
```

Comment:
why not use it as `computeContainerForBoundText` and pass height/width as needed, 
as both functions are functionally doing the same

Reply:
also you can short things out `boundTextElementWidth/Height + BOUND_TEXT_PADDING * 2` is **common**, you can pre-calculate it before and use it down eleswhere

Reply:
you can short the code little more `boundTextElementWidth/Height + BOUND_TEXT_PADDING * 2` is mostly **common**, you can per-calculate and use it down elsewhere